src/main/java/com/abramchik/taskTwoCollections/linkedList/MyLinkedListImpl.java

```diff
@@ -0,0 +1,203 @@
+                lastNode = currentNode;
+            }
+        } else {
+            firstNode = new Node<>(t, null, null);
+            if (lastNode == null) {
+                lastNode = firstNode;
+            }
+        }
+        size++;
+    }
+
+    @Override
+    public int size() {
+        return size;
+    }
+
+    @Override
+    public T getElement(int index) {
+        return findNode(index).getCurrentElement();
+    }
+
+    @Override
+    public void clear() {
+        firstNode = null;
+        lastNode = null;
+        size = 0;
+    }
+
+    @Override
+    public T getLast() {
+        return lastNode.getCurrentElement();
+    }
+
+    @Override
+    public T getFirst() {
+        return firstNode.getCurrentElement();
+    }
+
+    @Override
+    public void remove(int index) {
+        Node<T> node = findNode(index);
+        if (node.getNextElement() == null) {
+            lastNode = lastNode.getPreviousElement();
+            lastNode.setNextElement(null);
+            size--;
+        } else if (node.getPreviousElement() == null) {
+            firstNode = firstNode.getNextElement();
+            firstNode.setPreviousElement(null);
+            size--;
+        } else {
+            Node<T> nextElement = node.getNextElement();
+            Node<T> previousElement = node.getPreviousElement();
+            nextElement.setPreviousElement(previousElement);
+            previousElement.setNextElement(nextElement);
+            size--;
+        }
+    }
+
+    @Override
+    public void remove(T t) {
+        for (Node<T> i = firstNode; i != null; i = i.getNextElement()) {
+            if (i.getCurrentElement().equals(t)) {
+                removeNode(i);
+            }
+        }
+    }
+
+    @Override
+    public Iterator<T> iterator() {
+        return new Iterator<>() {
+            int counter = 0;
+
+            @Override
+            public boolean hasNext() {
+                return counter < size;
+            }
+
+            @Override
+            public T next() {
+                return getElement(counter++);
+            }
+        };
+    }
+
+    @Override
+    public Iterator<T> descendingIterator() {
+        return new Iterator<>() {
+            int counter = size - 1;
+
+            @Override
+            public boolean hasNext() {
+                return counter >= 0;
+            }
+
+            @Override
+            public T next() {
+                return getElement(counter--);
+            }
+        };
+    }
+
+    private Node<T> findNode(int index) {
+        if (index == 0) {
+            return firstNode;
+        } else if (index == (size - 1)) {
+            return lastNode;
+        } else if (index > 0 && index < size - 1) {
+            Node<T> node = firstNode;
+            for (int i = 1; i <= index; i++) {
+                node = node.getNextElement();
+            }
+            return node;
+        } else throw new IndexOutOfBoundsException();
+    }
```

Comment:
Always use brackets with if, else etc
Avoiding curly brackets can allow to bugs :
https://www.imperialviolet.org/2014/02/22/applebug.html
Java code convention, 7.4:
https://www.oracle.com/java/technologies/javase/codeconventions-statements.html

Reply:
I think it is allowed, because there is one line, but I try to put brackets everywhere, I fixed it, and added HashMap)